### PR TITLE
docs(CDN): fix and supplement CDN domain resource documentation

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:

- fix and supplement CDN domain resource documentation

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [ ] Tests added/passed.
* [X] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```

```

### CODE REVIEW

File: docs/resources/cdn_domain.md [hunk 1]
```diff
@@ -300,12 +355,16 @@ The `rules` block support:
...
+* `content` - (Optional, String) Specifies the content that matches `rule_type`.
+  + If `rule_type` is set to **all** or **home_page**, keep this parameter empty.
+  + If `rule_type` is set to **file_extension**, the value of this parameter is a list of file name
   extensions. A file name extension starts with a period (.). File name extensions are separated by semicolons (;),
-  for example, .jpg;.zip;.exe. If `rule_type` is set to **2**, the value of this parameter is a list of directories.
-  A directory starts with a slash (/). Directories are separated by semicolons (;), for example,
-  /test/folder01;/test/folder02.
+  for example, `.jpg;.zip;.exe`. Up to 20 file types are supported.
+  + If `rule_type` is set to **catalog**, the value of this parameter is a list of directories. A directory starts with
+  a slash (/). Directories are separated by semicolons (;), for example, `/test/folder01;/test/folder02`.
```
Comment (jinyangyang222) [inline docs/resources/cdn_domain.md:364]: Indent four spaces

File: docs/resources/cdn_domain.md [hunk 2]
```diff
@@ -211,26 +240,44 @@ The `https_settings` block support:
...
-* `value` - (Optional, String) Specifies the value of request or response header.
+<a name="http_response_header_object"></a>
+The `http_response_header` block support:
+
+* `name` - (Required, String) Specifies the HTTP response header. Valid values are `Content-Disposition`, `Content-Language`,
+  `Access-Control-Allow-Origin`, `Access-Control-Allow-Methods`, `Access-Control-Max-Age`, `Access-Control-Expose-Headers`,
+  `Access-Control-Allow-Headers` or custom headers. A header contains `1` to `100` characters, including letters, digits,
+  and hyphens (-), and starts with a letter.
+
+* `action` - (Required, String) Specifies the operation type of the HTTP response header. The value can be `set` or `delete`.
```
Comment (jinyangyang222) [inline docs/resources/cdn_domain.md:277]:
```suggestion
* `action` - (Required, String) Specifies the operation type of the HTTP response header. The value can be **set** or **delete**.
```

**Review by jinyangyang222 — COMMENTED** [review]
Error in indenting four spaces and using double asterisks to emphasize text, please check all

File: docs/resources/cdn_domain.md [hunk 3]
```diff
@@ -121,46 +121,72 @@ resource "huaweicloud_cdn_domain" "domain_1" {
...
+  Domain names at all levels can only be composed of letters, digits, and hyphens (-), and the letters are equivalent in
+  upper and lower case. Domain names at all levels are connected with (.). The domain name can contain up to `75` characters.
+
+* `type` - (Required, String, ForceNew) Specifies the service type of the domain name. Changing this parameter will
+  create a new resource. The valid values are as follows:
+  + **web**: Static acceleration. For websites with many images and small files, such as portals and e-commerce websites.
+  + **download**: Download acceleration. For large files, such as apps in app stores and game clients.
+  + **video**: Streaming media acceleration. For video on demand (VOD) websites and online education websites.
+  + **wholeSite**: Whole site acceleration. For websites with both dynamic and static content, such as online exam
+  platforms, forums, and blogs.
```
[the remark this answers is not in the document]
Comment (deer-hang) [reply docs/resources/cdn_domain.md]: OK

File: docs/resources/cdn_domain.md (hunk 1 above)
Comment (deer-hang) [reply docs/resources/cdn_domain.md]: OK

File: docs/resources/cdn_domain.md (hunk 2 above)
Comment (deer-hang) [reply docs/resources/cdn_domain.md]: OK

**Pull request merged**